service/src/main/kotlin/io/provenance/explorer/service/NftService.kt

```diff
@@ -31,39 +41,121 @@ class NftService(
     private val protoPrinter: JsonFormat.Printer
 ) {
 
+    fun getScopeDescrip(addr: String) =
+        metadataClient.getScopeSpecById(addr).scopeSpecification.specification.description
+
     fun getAllScopes(page: Int, count: Int) =
         metadataClient.getAllScopes(page.toOffset(count), count).let { res ->
-            val uuids = res.scopesList.map { it.scopeIdInfo.scopeUuid }
-            PagedResults(res.pagination.total.pageCountOfResults(count), uuids)
+            val rows = res.scopesList.map { s ->
+                val lastTx = TxNftJoinRecord.findTxByUuid(s.scopeIdInfo.scopeUuid, 0, 1).firstOrNull()
+                ScopeListview(
+                    s.scopeIdInfo.scopeAddr,
+                    getScopeDescrip(s.scopeSpecIdInfo.scopeSpecAddr)?.name,
+                    s.scopeSpecIdInfo.scopeSpecAddr,
+                    lastTx?.txTimestamp?.toString() ?: ""
+                )
+            }
+            PagedResults(res.pagination.total.pageCountOfResults(count), rows)
         }
 
     fun getScopesForOwningAddress(address: String, page: Int, count: Int) =
         metadataClient.getScopesByOwner(address, page.toOffset(count), count).let {
             PagedResults(it.pagination.total.pageCountOfResults(count), it.scopeUuidsList)
         }
 
-    fun getScopeByUuid(uuid: String) = metadataClient.getScopeById(uuid)
-        .let { it.scope.scopeIdInfo.scopeUuid to it.recordsList.map { rec -> rec.recordIdInfo.recordAddr } }
+    fun getScopeByAddr(addr: String) = metadataClient.getScopeById(addr)
+        .let {
+            val spec = getScopeDescrip(it.scope.scopeSpecIdInfo.scopeSpecAddr)
+            ScopeDetail(
+               it.scope.scopeIdInfo.scopeAddr,
+                spec?.name,
+                it.scope.scopeSpecIdInfo.scopeSpecAddr,
+                spec?.toSpecDescrip(),
+                it.scope.scope.ownersList.toOwnerRoles(),
+                it.scope.scope.valueOwnerAddress
+            )
+        }
+
+    fun getRecordsForScope(addr: String): List<ScopeRecord> {
+        // get scope
+        val scope = metadataClient.getScopeById(addr, true)
+        // get scope spec -> contract specs
+        val scopeSpec = metadataClient.getScopeSpecById(scope.scope.scopeSpecIdInfo.scopeSpecAddr)
+        val contractSpecs = scopeSpec.scopeSpecification.specification.contractSpecIdsList.map { it.toMAddress() }
+        // get record specs for each contract spec
+        val recordSpecs = contractSpecs
+            .flatMap { cs -> metadataClient.getRecordSpecsForContractSpec(cs.toString()).recordSpecificationsList }
+            .groupBy { it.specification.name }
+        // get records/sessions
+        val records = scope.recordsList.associateBy { it.record.name }
+        val sessions = scope.sessionsList.associateBy { it.sessionIdInfo.sessionAddr }
+        // map record specs
+        // apply info from records
+        val list = recordSpecs.map { (k, v) ->
```

Comment:
Not sure if you want to leave the comments in the code here?  I know these were probably there to declare the steps that needed to be implemented.

Reply:
Re did comments due to some complexity in the code.